facet-core/src/impls_alloc/vec.rs

```diff
@@ -3,6 +3,8 @@ use core::hash::Hash as _;
 
 use alloc::vec::Vec;
 
+type VecIterator<'mem, T> = core::slice::Iter<'mem, T>;
+
```

Comment:
I added this type alias to ensure that all the methods in `iter_vtable` align when casting from/to `PtrMut`